### ISSUE TITLE
c_windows_dbg_native.cmake/build/Debug/h2_sockpair_1byte_nosec_test.exe max_concurrent_streams failure

### ISSUE THREAD

**Issue by markdroth** [grpc/grpc#12388]
As seen in #12298:

https://grpc-testing.appspot.com/job/gRPC_pr_win/7476/testReport/junit/(root)/c_windows_dbg_native/cmake_build_Debug_h2_sockpair_1byte_nosec_test_exe_max_concurrent_streams/

Here's the output:

```
D0901 16:18:10.962000000  4088 test_config.c:388] test slowdown factor: sanitizer=1, fixture=2, poller=1, total=2
D0901 16:18:10.963000000  4088 dns_resolver.c:301] Using native dns resolver
I0901 16:18:10.963000000  4088 max_concurrent_streams.c:38] Running test: test_max_concurrent_streams_with_timeout_on_first/chttp2/socketpair_one_byte_at_a_time
I0901 16:18:16.977000000  4088 max_concurrent_streams.c:38] Running test: test_max_concurrent_streams_with_timeout_on_second/chttp2/socketpair_one_byte_at_a_time
I0901 16:18:22.985000000  4088 max_concurrent_streams.c:38] Running test: test_max_concurrent_streams/chttp2/socketpair_one_byte_at_a_time
E0901 16:18:22.992000000  4088 cq_verifier.c:273] cq returned unexpected event: OP_COMPLETE: tag:00000192 OK
E0901 16:18:22.992000000  4088 cq_verifier.c:280] expected tags:
tag(102) GRPC_OP_COMPLETE success=1 C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_dbg_native\test\core\end2end\tests\max_concurrent_streams.c:378

Abort handler called.
*** 0: 00000000001DC5CF print_current_stack - 00000000001DC530
*** 1: 00000000001DCB4D abort_handler - 00000000001DCB20
*** 2: 000000006ED0153B raise - 000000006ED011D0
*** 3: 000000006ED02B12 abort - 000000006ED02AE0
*** 4: 000000000015DA70 cq_verify - 000000000015D8A0
*** 5: 000000000010C3AB test_max_concurrent_streams - 000000000010B680
*** 6: 000000000010E744 max_concurrent_streams - 000000000010E6C0
*** 7: 00000000000E6344 grpc_end2end_tests - 00000000000E5090
*** 8: 00000000000E4D6E main - 00000000000E4CE0
*** 9: 00000000001F6ABE invoke_main - 00000000001F6AA0
*** 10: 00000000001F690A __scrt_common_main_seh - 00000000001F67B0
*** 11: 00000000001F679D __scrt_common_main - 00000000001F6790
*** 12: 00000000001F6AD8 mainCRTStartup - 00000000001F6AD0
*** 13: 00000000765D336A BaseThreadInitThunk - 00000000765D3358
*** 14: 00000000774A9902 RtlInitializeExceptionChain - 00000000774A989F
*** 15: 00000000774A98D5 RtlInitializeExceptionChain - 00000000774A989F
```

Yash, is this a similar issue to #12201?

**Comment by yashykt** [comment]
Yes, this is the same issue. The server is rejecting the second stream.

**Comment by yashykt** [comment]
Fixed in #12463
